centinel/experiments/baseline.py

```diff
@@ -0,0 +1,271 @@
+            self.record_pcaps = False
+        else:
+            self.record_pcaps = True
+
+        if os.geteuid() != 0:
+            logging.info("Centinel is not running as root, "
+                         "traceroute will be limited to UDP.")
+            self.traceroute_methods = ["udp"]
+        else:
+            # if running as root, TCP and UDP traceroute
+            # doing all 3 methods takes a lot of time
+            # self.traceroute_methods = ["icmp", "udp", "tcp"]
+            self.traceroute_methods = ["tcp", "udp"]
+
+    def run(self):
+        if self.record_pcaps:
+            self.external_results = {}
+
+        for input_file in self.input_files.items():
+            logging.info("Testing input file %s..." % (input_file[0]))
+            self.results.append(self.run_file(input_file))
+
+    def run_file(self, input_file):
+        file_name, file_contents = input_file
+
+        # Initialize the results for this input file.
+        # This can be anything from file name to version
+        # to any useful information.
+        result = {}
+        result["file_name"] = file_name
+
+
+        http_results = {}
+        tls_results = {}
+        dns_results = {}
+        traceroute_results = {}
+        url_metadata_results = {}
+        file_metadata = {}
+        file_comments = []
+
+        # each pcap file is stored in a separate file
+        # designated by a number. the indexes are stored
+        # in the json file and the pcap files are stored
+        # with their indexes as file names.
+        pcap_results = {}
+        pcap_indexes = {}
+        url_index = 0
+        comments = ""
+
+        # we may want to make this threaded and concurrent
+        for line in file_contents:
+            line = line.strip()
+
+            # parse file comments, if it looks like "key : value",
+            # parse it as a key-value pair. otherwise, just
+            # store it as a raw comment.
+            if line[0] == '#':
+                line = line[1:].strip()
+                if len(line.split(':')) > 1:
+                    key, value = line.split(':', 1)
+                    key = key.strip()
+                    value = value.strip()
+                    file_metadata[key] = value
+                else:
+                    file_comments.append(line)
+                continue
+
+            url = line
+            meta = ''
+            url_index = url_index + 1
+
+            # handle cases where URL is enclosed in quotes
+            if line[0] == '"':
+                next_quote = line[1:].find('"')
+                url = line[1:next_quote + 1]
+                # try to separate metadata, if any
+                try:
+                    meta = ','.join(line[next_quote + 2:].split(',')[1:])
+                except:
+                    pass
+
+            # if the line doesn't start with a quote
+            # but the list entry has comma separated meta-data
+            elif len(line.split(',')) > 1:
```

Comment:
you keep doing all of these checks. Your code would be much clearer if you made a few booleans to store whether or not it is a comment and read in the line once. You could use csv to read the line in, then just have a few sections where you process-> it would make your code more concise and cleaner